<a href="https://colab.research.google.com/github/dscoool/chungbuk/blob/main/tensorflow_beginner.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 파이썬 텐서플로우 시작하기!!

간단한 예제(tensorflow)를 아무 생각 없이 따라 돌려 봅시다. 😊

본인의 PC에서 주피터 노트북(JUPYTER NOTEBOOK tensorflow_beginner.ipynb파일)을 실행하실 분은 우측 상단의 'Download'버튼을 눌러 파일을 다운로드받으세요!!

PyCharm, Visual Studio Code, Anaconda 무엇이든 사용하셔도 좋습니다.

그럼 실습을 시작해 볼까요? 😊

tensorflow라는 모듈을 사용할게요!!


아래는 예제 설명이 되어 있는 블로그입니다.

* 예제 설명: https://guide.ncloud-docs.com/beta/docs/tensorflow-for-beginner


먼저 import 명령어를 사용하여, 프로그램에 텐서플로 라이브러리를 임포트합니다


In [1]:
import tensorflow as tf


[MNIST 데이터셋](https://brunch.co.kr/@minwoo/25) 을 로드하여 준비합니다. 샘플 값을 정수에서 실수(float, 부동소수)로 변환합니다:

In [2]:
#mnist 데이터를 자동으로 다운로드합니다.
mnist = tf.keras.datasets.mnist

# {x} 변수(variables) 버튼을 눌러서, mnist 변수 내용을 살펴봅시다!!
# 아니면 콘솔에 mnist 를 실행시켜 변수 내용을 살펴봅시다!!

# train data와 test data를 분할합니다!!


(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11490434/11490434 [==============================] - 0s 0us/step


In [3]:
# 데이터가 궁금하면 한 개씩 주석을 풀고 실행해 봅시다!!
# 어떤 데이터가 로드되었을까요?
# mnist.load_data()
# x_train
# x_test
# y_train
y_test

array([7, 2, 1, ..., 4, 5, 6], dtype=uint8)

기계학습의 'layer' (층) 개념을 배운 적이 있나요?

층을 차례대로 쌓아 tf.keras.Sequential 모델을 만듭니다. 훈련에 사용할 옵티마이저(optimizer)와 손실 함수를 선택합니다:

In [4]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

모델을 훈련하고 평가합니다:

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)


''' 예상 결과
Train on 60000 samples
Epoch 1/5
WARNING:tensorflow:Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'
WARNING: Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'
60000/60000 [==============================] - 4s 68us/sample - loss: 0.2941 - accuracy: 0.9140
Epoch 2/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.1396 - accuracy: 0.9587
Epoch 3/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.1046 - accuracy: 0.9680
Epoch 4/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.0859 - accuracy: 0.9742
Epoch 5/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.0724 - accuracy: 0.9771
10000/1 - 0s - loss: 0.0345 - accuracy: 0.9788
[0.06729823819857557, 0.9788]
'''

Epoch 1/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.2952 - accuracy: 0.9145
Epoch 2/5
1875/1875 [==============================] - 6s 3ms/step - loss: 0.1411 - accuracy: 0.9586
Epoch 3/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.1058 - accuracy: 0.9678
Epoch 4/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0868 - accuracy: 0.9738
Epoch 5/5
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0749 - accuracy: 0.9770
313/313 - 0s - loss: 0.0728 - accuracy: 0.9779 - 396ms/epoch - 1ms/step


" 예상 결과\nTrain on 60000 samples\nEpoch 1/5\nWARNING:tensorflow:Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'\nWARNING: Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'\n60000/60000 [==============================] - 4s 68us/sample - loss: 0.2941 - accuracy: 0.9140\nEpoch 2/5\n60000/60000 [==============================] - 4s 62us/sample - loss: 0.1396 - accur

훈련된 이미지 분류기는 이 데이터셋에서 약 98%의 정확도를 달성합니다. 더 자세한 내용은 TensorFlow 튜토리얼 을 참고해 주십시오.

* Reference: 텐서플로우 튜토리얼- https://www.tensorflow.org/tutorials

